Make Repro-CI Artifacts Reflect Configs Repo Structure

CodeGat opened the pull request:
## The Problem

Currently, the struture of the checksums in a repro test experiment running on Gadi are the following:

```
$EXPERIMENT_LOCATION/
└── checksum/
    ├── historical-*hr-checksum.json
    └── test_report.xml
```

And the structure of the checksums in the repo are:
```
.
└── testing/
    └── checksum/
        ├── historical-*hr-checksum.json
        └── test_report.xml
```
In which, in the CI, we special-case the logic for creating the intermediate `testing` directory in the repo. This also creates problems for the `inputs.additional-artifact-content-paths`, which are, confusingly, in the correct place at both the experiment level and the repo level, which means they both have to be handled differently. 

## The Solution

This PR unifies the checksum directories so that both the experiment and repo reflect the following structure:

```
$EXPERIMENT_LOCATION_OR_REPO_ROOT/
└── testing/
    └── checksum/
        ├── historical-*hr-checksum.json
        └── test_report.xml
```


In this PR:
- **test-repro.yml: put checksums under `./testing` directory in artifact and experiment to mimic default repo structure**
- **generate-checksums.yml: put checksums under `./testing` directory in artifact and experiment to mimic default repo structure**
- **Updated the creation and usage of artifacts to account for the new `./testing` directory for checksums**

**NOTE:** There is a unhealthy amount of duplication of effort in the PR, especially in `test-repro`/`generate-checksums` (see https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/34b769750da1a419fee47f7a1644c60c23c33364 and https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/37518f4d439b3c2b2b5e63fad5a3fa0a3f837307). This will be rectified in #72.

**NOTE:** This is required to merge https://github.com/ACCESS-NRI/model-config-tests/pull/73

Closes https://github.com/ACCESS-NRI/model-config-tests/issues/75

aidanheerdegen commented on .github/workflows/config-pr-1-ci.yml:
This comment needs updating I think?

https://github.com/ACCESS-NRI/model-config-tests/pull/74/files#diff-5cced7dbde2dd0758da5d65e986beb12cd2aba81deb1868c64d94846a49af26dL21

CodeGat replied:
Done in https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/ac85a231f87f12d7684c6962673d1df2d45b30ba

aidanheerdegen commented on .github/workflows/generate-checksums.yml:
```diff
@@ -181,10 +183,10 @@ jobs:
+      - name: Move Outputs to Repo
         run: |
           mkdir -p ${{ inputs.committed-checksum-location }}
-          mv ${{ env.OUTPUT_LOCAL_LOCATION }}/checksum/* ${{ inputs.committed-checksum-location }}
+          mv ${{ env.OUTPUT_LOCAL_LOCATION }} .
 
```
I think you either need to `cd` into `${{ inputs.committed-checksum-location }}` or `cp` into that location. No?

CodeGat replied:
I realised that we don't actually need that input anymore - `testing/checksum` is assumed in a lot of places. Removed in https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/539f893e3cc52c404e5a31d3745f7b3bd042b1fe

aidanheerdegen reviewed (commented):
I accidentally just commented directly, but I have a few questions.

CodeGat commented:
Alright, there's been a bit of change since the last review @aidanheerdegen - this is because I realised that in `OM3`s case, we needed to an additional artifact path that has `docs` prepended. Here are those changes in a more easily digestable format:

-  Added `inputs.additional-artifact-content-paths-root` to [`test-repro`](https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/8ad1c71f1534233d0bed8a76e219b0756f75a59f) and [`config-pr-1-ci.yml`](https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/ddd7cfcb9309eefe042aff41013bdbb7faa75a5f)
- Added `inputs.additional-artifact-content-paths-commit` to [`config-pr-2-confirm.yml`](https://github.com/ACCESS-NRI/model-config-tests/pull/74/commits/3e9ad756227474e6763b43ad6863ca8ef8226218)
- Along with the fixes that you had noted earlier :D

CodeGat commented:
Closing this for now as it deals with committing `inputs.additional-artifact-content-paths` which we don't want to enable just yet. 
It still has a lot of good stuff relating to making the artifacts closer to the experiment, so I won't delete the branch just yet.
